packages/metro/src/node-haste/DependencyGraph.js

```diff
@@ -133,7 +133,7 @@ class DependencyGraph extends EventEmitter {
       retainAllFiles: true,
       resetCache: config.resetCache,
       rootDir: config.projectRoot,
-      roots: config.watchFolders,
+      roots: config.roots,
```

Comment:
If we specify both `roots` and `watchFolders`, `roots` will be watched and `watchFolders` won't be. Is this what you intended?

Reply:
My intent was to build the haste map around the roots so they'd be found by the resolver. I thought watching happened elsewhere. I didn't know the haste map also handled watching.

It sounds like this needs to go deeper, then, into the haste map code. Do you have pointers/advice on how I can expand the haste map library to support this? I haven't looked at it before.